packages/hooks/src/useTrackedEffect/index.ts

```diff
@@ -1,21 +1,21 @@
 import { useEffect, useRef, DependencyList } from 'react';
 
 const useTrackedEffect = (effect, deps?: DependencyList) => {
-  const initDeps = useRef<DependencyList>();
+  const previousDepsRef = useRef<DependencyList>();
   const diffTwoDeps = (deps1, deps2) => {
     //Let's do a reference equality check on 2 dependency list.
     //If deps1 is defined, we iterate over deps1 and do comparison on each element with equivalent element from deps2
     //As this func is used only in this hook, we assume 2 deps always have same length.
     return deps1
-      ? deps1.map((_ele, idx) => (deps1[idx] != deps2[idx] ? idx : -1)).filter((ele) => ele >= 0)
+      ? deps1.map((_ele, idx) => (deps1[idx] !== deps2[idx] ? idx : -1)).filter((ele) => ele >= 0)
       : deps2
       ? deps2.map((_ele, idx) => idx)
       : [];
   };
   useEffect(() => {
-    let changes = diffTwoDeps(initDeps.current, deps);
-    initDeps.current = deps;
-    return effect(changes);
+    let changes = diffTwoDeps(previousDepsRef.current, deps);
+    previousDepsRef.current = deps;
+    return effect(changes, previousDepsRef.current, deps);
```

Comment:
We need to get the old value before setting `previousDeps.current`

```suggestion
    const previousDeps = previousDepsRef.current;
    previousDepsRef.current = deps;
    return effect(changes, previousDeps, deps);
```